webpack.config.js

```diff
@@ -1,83 +1,108 @@
-const isProduction = NODE_ENV === 'production'
+const isDevelopment = NODE_ENV === "development"
+const isStaging = NODE_ENV === "staging"
+const isProduction = NODE_ENV === "production"
 const isDeploy = isStaging || isProduction
 
+const cacheDirectory = path.resolve(__dirname, ".cache")
+
 const config = {
-  devtool: WEBPACK_DEVTOOL || 'cheap-module-source-map',
+  devtool: WEBPACK_DEVTOOL || "cheap-module-source-map",
   entry: {
     webpack: [
-      'webpack-hot-middleware/client?reload=true',
-      './src/desktop/apps/webpack/client.js',
+      "webpack-hot-middleware/client?reload=true",
+      "./src/desktop/apps/webpack/client.js",
     ],
     ...getEntrypoints(),
   },
   output: {
-    filename: '[name].js',
-    path: path.resolve(__dirname, 'public/assets'),
-    publicPath: '/assets',
-    sourceMapFilename: '[file].map?[contenthash]',
+    filename: "[name].js",
+    path: path.resolve(__dirname, "public/assets"),
+    publicPath: "/assets",
+    sourceMapFilename: "[file].map?[contenthash]",
   },
   module: {
     rules: [
       {
         test: /\.coffee$/,
         include: /src/,
-        use: [{ loader: 'coffee-loader' }],
+        use: [
+          {
+            loader: "cache-loader",
+            options: {
+              cacheDirectory: path.join(cacheDirectory, "coffee"),
+            },
+          },
+          { loader: "coffee-loader" },
+        ],
       },
       {
         test: /\.(jade|pug)$/,
         include: /src/,
-        loader: 'pug-loader',
-        options: {
-          doctype: 'html',
-          root: __dirname,
-        },
+        use: [
+          {
+            loader: "cache-loader",
+            options: {
+              cacheDirectory: path.join(cacheDirectory, "pug"),
+            },
+          },
+          {
+            loader: "pug-loader",
+            options: {
+              doctype: "html",
+              root: __dirname,
+            },
+          },
+        ],
       },
       {
         test: /(\.(js|ts)x?$)/,
-        include: path.resolve('./src'),
+        include: path.resolve("./src"),
         use: [
-          { loader: 'cache-loader' },
           {
-            loader: 'babel-loader',
-            query: {
-              cacheDirectory: true,
+            loader: "babel-loader",
+            options: {
+              cacheDirectory: path.join(cacheDirectory, "babel"),
             },
           },
         ],
       },
       {
         test: /\.json$/,
-        loader: 'json-loader',
+        loader: "json-loader",
       },
     ],
   },
   plugins: [
     // TODO: Add webpack typechecker
-    new ProgressBarPlugin(),
-
+    new SimpleProgressWebpackPlugin({ format: "compact" }),
     // Remove moment.js localization files
     new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
-
+    new HardSourceWebpackPlugin({
+      cacheDirectory: path.join(cacheDirectory, "hard-source"),
+      info: {
+        mode: "none",
+        level: "error",
+      },
+    }),
```

Comment:
I've tried this plugin before but noticed that it seemed to slow things down a bit, as opposed to speeding things up -- was probably misconfigured in some way.

Reply:
The initial definitely takes a bit longer but subsequent builds are like night and day.